bfss3/bfss3.go

```diff
@@ -385,10 +390,11 @@ func (r *response) Read(p []byte) (n int, err error) {
 // --------------------------------------------------------------------
 
 type iterator struct {
-	parent  *bucket
-	ctx     context.Context
-	pattern string
-	token   *string
+	ctx context.Context
+
+	bucket    interface{ stripPrefix(string) string }
```

Comment:
why? there is no reason to do this here. (pointers to) interfaces add additional weight and use more memory. this is a private struct, there is no need to do this.